### PR TITLE
Fixed #600: Empty Note error

### LINKED ISSUE #600: 🐛 [Bug report]: Empty Note error
**Describe the bug**
The user has the ability to make blank notes. If the `Note Title` and `Note Content` are both blank, no user should be able to create an empty note.

**To Reproduce**
1. Go to the [Keeper App](https://reactplay.io/plays/shivam-katare/keeper)
2. Click on the `+` button keeping both `Note Title`  and `Note Content` empty.

**Expected behavior**
Users shouldn't be able to create any empty notes and should instead receive some sort of popup notification asking them to "Please provide some valid data."

**Screenshots**
Screenshot 2022-10-04 at 12 31 50 AM

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: Redmi K20 pro
 - OS: Android
 - Browser: Chrome

I will be more than happy to fix it. Let me know if this makes sense.

### PR DESCRIPTION
**Pull request by siddxharth**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Fix the issue where the user could create a new note with an empty title and note body. Now the button is disabled if the title and body fields both are empty.

Fixes # (issue)

#600 

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

I tested these changes myself by running the server locally with the credentials provided in the README file.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by Angryman18** [comment]
Screenshot from 2022-10-11 07-52-21
can you fix this '+' button also. and make it center

**Comment by siddxharth** [comment]
Okay, I'll fix it

**Comment by siddxharth** [comment]
I've made a new pull request. Made a couple of changes...
1. The Create button is now centred.
2. The note title and content will be considered empty even if it only contains spaces.

File: src/plays/keeper/CreateArea.jsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ function CreateArea(props) {
           value={note.content}
           placeholder="Take a note..."
         />
-        <button disabled={note.title === '' && note.content === ''} className="create-button" onClick={submitNote}>< FaPlusCircle /></button>
+        <button disabled={note.title.trim().length === 0 && note.content.trim().length === 0} className={(note.title.trim().length === 0 && note.content.trim().length === 0) ? 'create-button-disabled' : 'create-button'} onClick={submitNote}>< FaPlusCircle /></button>
```
Comment (Angryman18) [inline src/plays/keeper/CreateArea.jsx:46]: as from readability prospective assign these expression to a variable and use them here.

**Review by Angryman18 — CHANGES_REQUESTED** [review]

File: src/plays/keeper/CreateArea.jsx (hunk 1 above)
Comment (siddxharth) [reply src/plays/keeper/CreateArea.jsx]: I've shifted these to new variables.

File: src/plays/keeper/CreateArea.jsx [hunk 2]
```diff
@@ -6,6 +6,8 @@ function CreateArea(props) {
     title: "",
     content: ""
   });
+  const title_len = note.title.trim().length;
+  const content_len = note.content.trim().length;
```
Comment (Angryman18) [inline src/plays/keeper/CreateArea.jsx:10]: its good but we can shorten it one more way like
`const disableButton =  note.title.trim().length === 0 && note.content.trim().length === 0`
or in short
`const disableButton =  !note.title.trim() && !note.content.trim()`

File: src/plays/keeper/CreateArea.jsx [hunk 3]
```diff
@@ -43,7 +45,7 @@ function CreateArea(props) {
           value={note.content}
           placeholder="Take a note..."
         />
-        <button disabled={note.title.trim().length === 0 && note.content.trim().length === 0} className={(note.title.trim().length === 0 && note.content.trim().length === 0) ? 'create-button-disabled' : 'create-button'} onClick={submitNote}>< FaPlusCircle /></button>
+        <button disabled={title_len === 0 && content_len === 0} className={(title_len === 0 && content_len === 0) ? 'create-button-disabled' : 'create-button'} onClick={submitNote}>< FaPlusCircle /></button>
```
Comment (Angryman18) [inline src/plays/keeper/CreateArea.jsx:48]: `<button disabled={disableButton} className={disableButton ? 'create-button-disabled' : 'create-button'} onClick={submitNote}>< FaPlusCircle /></button>`

Comment (Angryman18) [reply src/plays/keeper/CreateArea.jsx:48]: do you thhink its good to have this changes?

**Review by Angryman18 — CHANGES_REQUESTED** [review]

File: src/plays/keeper/CreateArea.jsx (hunk 3 above)
Comment (siddxharth) [reply src/plays/keeper/CreateArea.jsx:48]: Yes, this is even simpler

**Comment by siddxharth** [comment]
I've made a new commit, it should be good now.

**Review by Angryman18 — APPROVED** [review]
LGTM

**Pull request merged**